pkg/function/scale_workload.go

```diff
@@ -115,14 +113,27 @@ func (s *scaleWorkloadFunc) ExecutionProgress() (crv1alpha1.PhaseProgress, error
 	}, nil
 }
 
-func getArgs(tp param.TemplateParams, args map[string]interface{}) (namespace, kind, name string, replicas int32, waitForReady bool, err error) {
+type scaleArgs struct {
+	namespace    string
+	kind         string
+	name         string
+	replicas     int32
+	waitForReady bool
```

Comment:
Should we make these fields part of `scaleWorkloadFunc`? Like we have `progressPercent`?

Reply:
The rule here is: We should not return these many variables. Since scaleWorkloadFunc.Exec is calling `getArgs()`, and `getArgs()` is called in this function only, do you think it would make sense to move it under `scaleWorkloadFunc.getArgs()`? These variables are usable in `Exec` function only, and not others.

Reply:
The arguments are properties of the function only, right? We should not make decisions based on whether they are used in one function or multiple. It also makes sense from an extensibility point of view, they can be referred to in other functions if required.

Reply:
I have moved them to `scaleWorkloadFunc.setArgs()`.